mobile_wallet/src/lib.rs

```diff
@@ -163,6 +218,182 @@ fn create_encrypted_transfer_aux(input: &str) -> anyhow::Result<String> {
+                .receive
+                .get(entrypoint_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find receive schema"))?
+                .parameter();
+            match entrypoint_parameter {
+                Some(value) => value.clone(),
+                None => return Err(anyhow::anyhow!("Missing parameter for entrypoint")),
+            }
+        }
+    };
+    Ok(receive_schema)
+}
+
+fn get_parameters_as_json(
+    payload: UpdateContractPayload,
+    schema: &str,
+    schema_version: &Option<u8>,
+) -> anyhow::Result<Value> {
+    let mut schema_bytes = hex::decode(schema)?;
+
+    let contract_name = &payload
+        .receive_name
+        .as_receive_name()
+        .contract_name()
+        .to_string();
+    let entrypoint_name = &payload
+        .receive_name
+        .as_receive_name()
+        .entrypoint_name()
+        .to_string();
+
+    let receive_schema = match from_bytes::<VersionedModuleSchema>(&schema_bytes) {
+        Ok(versioned) => get_receive_schema(versioned, contract_name, entrypoint_name)?,
+        Err(_) => match schema_version {
+            Some(version) => {
+                let mut versioned_schema_bytes: Vec<u8> = vec![u8::MAX, u8::MAX, *version];
+                versioned_schema_bytes.append(&mut schema_bytes);
+                let versioned_schema: VersionedModuleSchema =
+                    match from_bytes(&versioned_schema_bytes) {
+                        Ok(v) => v,
+                        Err(_) => return Err(anyhow::anyhow!("Invalid schema or schema version")),
+                    };
+                get_receive_schema(versioned_schema, contract_name, entrypoint_name)?
+            }
+            None => return Err(anyhow::anyhow!("Missing schema version")),
+        },
+    };
+
+    let parameter_bytes: Vec<u8> = payload.message.into();
+    let mut parameters_cursor = Cursor::new(&parameter_bytes[..]);
+    match receive_schema.to_json(&mut parameters_cursor) {
+        Ok(schema) => Ok(schema),
+        Err(e) => Err(anyhow::anyhow!(
+            "Unable to parse parameters to JSON: {:?}",
+            e
+        )),
+    }
+}
+
+fn transaction_to_json_aux(input: &str) -> anyhow::Result<String> {
+    let v: Value = from_str(input)?;
+    let serialized_transaction: String = try_get(&v, "transaction")?;
+    let pre_tx: PreAccountTransaction = base16_decode_string(&serialized_transaction)?;
+
+    let response = match pre_tx.clone().payload {
+        Payload::Update { payload } => {
+            let schema: String = try_get(&v, "schema")?;
+            let schema_version: Option<u8> = maybe_get(&v, "schemaVersion")?;
+            let parameters_as_json = get_parameters_as_json(payload, &schema, &schema_version)?;
+            let transaction_as_json = serde_json::json!(pre_tx);
+            replace_json_value(
+                &transaction_as_json,
+                &String::from("message"),
+                &parameters_as_json,
+            )
```

Comment:
This seems a bit hacky and not in the spirit of the other code we have. Is it possible to instead return 
```
json!({
    "hashToSign": pre_tx.hash_to_sign,
    "header: pre_tx.header,
    "payload": {
        "update": {
        "amount": payload.amount,
        "address": payload.address,
        "receiveName": payload.receive_name,
        "message": parameters_as_json
       }
)
```
instead?

Reply:
I have changed it to follow your suggestion.